torch/distributed/algorithms/_comm_hooks/default_hooks.py

```diff
@@ -0,0 +1,126 @@
+import torch
+import torch.distributed as dist
+from torch.distributed import distributed_c10d
+
+
+class DefaultState(object):
+    r"""
+    Stores state needed to perform the default ``all_reduce`` algorithm
+    within a communication hook.
+
+    Args:
+        process_group (ProcessGroup): The process group to be used for all-reduce.
+        world_size (int): The number of workers in a process group.
+            Determined based on a ``process_group``.
+        gradient_predivide_factor (float): A factor for gradients' pre-division.
```

Comment:
I guess these aren't actually args into DefaultState, so we can remove them from docs?